js-apps/meep-frontend/src/js/containers/idc-map.js

```diff
@@ -0,0 +1,425 @@
+/*
+ * Copyright (c) 2019  InterDigital Communications, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { connect } from 'react-redux';
+import React, { Component, createRef } from 'react';
+import ReactDOM from 'react-dom';
+import L from 'leaflet';
+import 'mapbox-gl';
+import 'mapbox-gl-leaflet';
+import deepEqual from 'deep-equal';
+import { updateObject } from '../util/object-util';
+import {
+  execChangeMap
+} from '../state/exec';
+import {
+  uiExecChangeSandboxCfg
+} from '../state/ui';
+
+class IDCMap extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.thisRef = createRef();
+    this.configRef = createRef();
+  }
+
+  componentDidMount() {
+    this.createMap();
+  }
+
+  componentWillUnmount() {
+    this.destroyMap();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.sandbox !== this.props.sandbox) {
+      this.destroyMap();
+      this.createMap();
+      this.updateMarkers();
+    }
+  }
+
+  shouldComponentUpdate(nextProps) {
+    // Check for map size update
+    let width = this.thisRef.current.offsetWidth;
+    let height = this.thisRef.current.offsetHeight;
+    if ((width && this.width !== width) || (height && this.height !== height)) {
+      this.width = width;
+      this.height = height;
+      // console.log('Map view resized to: ' + width + 'x' + height);
+      this.map.invalidateSize();
+      return true;
+    }
+
+    // Update if sandbox changed
+    if (nextProps.sandbox !== this.props.sandbox) {
+      return true;
+    }
+
+    // Update if map changed
+    if (!deepEqual(nextProps.execMap, this.props.execMap)) {
+      return true;
+    }
+
+    return false;
+  }
+
+  createMap() {
+    // Get stored configuration
+    var cfg = this.props.sandboxCfg[this.props.sandboxName];
+    var lat = cfg.center ? cfg.center.lat : 43.73752;
+    var lng = cfg.center ? cfg.center.lng : 7.42892;
+    var zoom = cfg.zoom ? cfg.zoom : 15;
+    var baselayerName = cfg.baselayerName ? cfg.baselayerName : 'Positron';
+ 
+    // Create Map instance
+    var domNode = ReactDOM.findDOMNode(this);
+    this.map = L.map(domNode, {
+      center: [lat,lng],
+      zoom: zoom,
+      minZoom: 15,
+      maxZoom: 18,
+      drawControl: true
+    });
+
+    // Create GL Baselayers
+    var positronBaselayer = L.mapboxGL({style: 'http://10.3.16.105:30080/styles/positron/style.json'});
```

Comment:
For the 4 styles... don't use IP address/port directly.

Reply:
Fixed.